packages/api/src/external/commonwell/document/document-query-sandbox.ts

```diff
@@ -88,39 +92,44 @@ export async function sandboxGetDocRefsAndUpsert({
     requestId,
   });
 
-  for (const entry of convertibleDocs) {
-    const prevDocId = entry.docRef.id;
-    entry.docRef.id = uuidv7();
-    try {
-      await convertCDAToFHIR({
-        patient,
-        document: {
-          id: entry.docRef.id,
-          content: { mimeType: entry.docRef.content?.[0]?.attachment?.contentType },
-        },
-        s3FileName: entry.s3Info.key,
-        s3BucketName: entry.s3Info.bucket,
-        requestId,
-      });
-
-      const contained = entry.docRef.contained ?? [];
-      const containsPatient = contained.filter(c => c.resourceType === "Patient").length > 0;
-      if (!containsPatient) {
-        contained.push({
-          resourceType: "Patient",
-          id: patient.id,
-        });
-      }
-      entry.docRef.subject = {
-        type: "Patient",
-        reference: `Patient/${patient.id}`,
-      };
+  for (const entry of docsWithContent) {
+    const fileTitle = entry.docRef.content?.[0]?.attachment?.title;
+    if (!existingDocTitles.includes(fileTitle)) {
+      const prevDocId = entry.docRef.id;
+      entry.docRef.id = uuidv7();
+      try {
+        if (convertibleDocs.find(d => d.docRef.id === entry.docRef.id)) {
```

Comment:
just a note for your knowledge:

these kind of lookups:
- `existingDocTitles.includes(fileTitle)`
- `convertibleDocs.find(d => d.docRef.id === entry.docRef.id)`

.. are totally fine in this case, since the volume of docs in Sandbox is super low, but it scales horribly when you're dealing with thousands, or millions, of items

ie you'd be potentially looping through a million docs, and for each one, looping through another million docs, just to see if the doc is convertible - this would make your algo O(N^2)

instead, a much more efficient solution would be to put all the convertible doc ids in a hashmap, and then do the lookup to see if the is exists in the map, reducing the complexity of your algo to O(N), since the lookup in the map is O(1)